### PR TITLE
converted lumen api to laravel

### PR DESCRIPTION
**Pull request by SalarCloudways**

### CODE REVIEW

File: app/Repositories/AuthorRepository.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    //Create New Author
+    public function createAuthors($data)
+    {
+        return new AuthorResource(Author::create($data->all()));
+    }
+
+    //Update an Author
+    public function updateAuthor($data, $id)
+    {
+        $author = new AuthorResource(Author::findOrFail($id));
```
Comment (faizytech) [inline app/Repositories/AuthorRepository.php]: we will not put author resource here

File: app/Models/Author.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+class Author extends Model
+{
+    use HasFactory;
+
+    protected $fillable = [
+        'name', 'email', 'github', 'twitter'
+    ];
+
+    public static $rules = [
```
Comment (faizytech) [inline app/Models/Author.php]: delete rules from here.

File: routes/web.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    //Get Single Author
+    Route::get('/authors/{author}', function(Author $author){
+        return new AuthorResource($author);
+    });
+
+    //Delete Author
+    Route::delete('/authors/{author}', function(Author $author){
+
+        $author->delete();
+        return Response("Author Deleted Successfully", 200);
```
Comment (faizytech) [inline routes/web.php]: This thing should be done in controller.

File: routes/web.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    Route::get('/authors', 'allauthors');
+
+    //Create new Author
+    Route::post('/authors', 'createauthor');
+
+    //Update an Author
+    Route::put('/authors/{id}', 'updateAuthor');
+
+    //Get Single Author
+    Route::get('/authors/{author}', function(Author $author){
```
Comment (faizytech) [inline routes/web.php]: It should be done in controller

**Review by faizytech — CHANGES_REQUESTED** [review]
Little changes

File: app/Repositories/AuthorRepository.php (hunk 1 above)
Comment (SalarCloudways) [reply app/Repositories/AuthorRepository.php]: fixed, using author resouce on return statement now

File: app/Models/Author.php (hunk 2 above)
Comment (SalarCloudways) [reply app/Models/Author.php]: added rules in separate file

File: routes/web.php (hunk 3 above)
Comment (SalarCloudways) [reply routes/web.php]: delete author in controller now

File: routes/web.php (hunk 4 above)
Comment (SalarCloudways) [reply routes/web.php]: showing single author from controller now

File: app/Models/Author.php (hunk 2 above)
Comment (faizytech) [reply app/Models/Author.php]: delete rules from this model. i can still see it here

**Review by faizytech — CHANGES_REQUESTED** [review]

**Review by faizytech — APPROVED** [review]
Kindly merge it.

**Pull request merged**